### DISCUSSION TITLE
GLM3是否有frequency_penalty, presence_penalty或类似的参数？

### DISCUSSION THREAD

**Discussion by estherche113** [zai-org/ChatGLM3#1173] [Q&A] [answered]
ChatGLM3-6b-32k 在 RAG (Retrieval Augmented Generation) 数据查询应用中，模型会自行对问题进行延伸解释，答案来源于训练数据而不是 prompt 或 retrieval 的数据。由于通过Prompt限制回答范围并不奏效，我们需要通过调整惩罚系数来避免模型输出多余信息。

OpenAI模型提供了frequency_penalty和presence_penalty两个参数，用来控制输出重复词汇/新词汇的概率。参考链接：https://community.openai.com/t/difference-between-frequency-and-presence-penalties/2777

我知道ChatGLM3有repetition_penalty，但它并没能起到我想要的效果。提高repetition_penalty减少了延伸回答的情况，但有更高概率出现乱码和编造的链接。

我想知道这个问题如何解决？repetition_penalty是如何作用的，它们之间有什么联系和区别？

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
有这个参数，调整到1.2
